app/controllers/events_controller.rb

```diff
@@ -96,4 +120,58 @@ def date_from_form(date_info)
     def event_params
       params.require(:event).permit(:name, :description, :max_participants, :active, :kind, :organizer, :knowledge_level)
     end
+
+    # Generate all names to print from the query-params
+    #
+    # @return participant_names as array of strings
+    def badges_name_params
+      participant_names = []
```

Comment:
Update: you can avoid parsing the key altogether if you pass the participant's name as value. See the comment to `app/views/events/badges.html.erb` :wink: